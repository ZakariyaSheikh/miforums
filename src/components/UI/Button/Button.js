import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ onClick, children = null, classNames = 'btn', ...rest }) => {
    return (
        <button className={classNames} onClick={onClick} type={rest.type}>
            {children}
        </button>
    );
};

Button.propTypes = {
    children: PropTypes.node,
    onClick: PropTypes.func.isRequired,
    classNames: PropTypes.string,
};
export default Button;
