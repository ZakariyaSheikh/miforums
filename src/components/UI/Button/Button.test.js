import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Button from './Button';

describe('</Button>', () => {
    const mockProps = {
        type: 'button',
        className: 'button-class',
        onClick: jest.fn(),
        children: 'Dummy Button',
    };

    test('renders as expected', () => {
        const wrapper = renderer.create(<Button {...mockProps} />).toJSON();
        expect(wrapper).toMatchSnapshot();
    });

    test('should call the onClick function when button clicked', () => {
        const wrapper = shallow(<Button {...mockProps} />);
        wrapper.find('button').simulate('click');
        expect(mockProps.onClick).toHaveBeenCalled();
    });

    test('should exists button node', () => {
        const wrapper = shallow(<Button {...mockProps} />);
        expect(wrapper.find('button')).toHaveLength(1);
    });

    test('should have the class btn rounded', () => {
        const wrapper = shallow(<Button {...mockProps} />);
        expect(wrapper.hasClass('button-class')).toBeTruthy();
    });

    test('should have a the default children', () => {
        const wrapper = shallow(<Button onClick={mockProps.onClick} />);
        expect(wrapper.find('button').text()).toBe('');
    });
    test('should have a children', () => {
        const wrapper = shallow(<Button {...mockProps} />);
        expect(wrapper.find('button').text()).toBe('Dummy Button');
    });
});
