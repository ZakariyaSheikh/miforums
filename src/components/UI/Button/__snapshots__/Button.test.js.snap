// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`</Button> renders as expected 1`] = `
<button
  className="button-class"
  onClick={[MockFunction]}
  type="button"
>
  Dummy Button
</button>
`;
